### PR TITLE
Lesson group new translation

### PR DESCRIPTION
**Pull request by dmcavoy**
[PLAT-52](https://codedotorg.atlassian.net/browse/PLAT-52)

Adds lesson group key-display_name pairs to scripts.en.yml for each script so that they will be translated.

Screen Shot 2020-04-20 at 8 05 23 PM

## Testing story

Updated script DSL tests to reflect changes.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/dsl/script_dsl_test.rb
Comment (davidsbailey) [inline dashboard/test/dsl/script_dsl_test.rb]: I had forgotten that these PL flex categories get used across scripts. I think this is ok, but here are the specific concerns that come to mind:

1. curriculum writers can now add their own lesson groups, it shouldn't be a problem for them to have to create a new lesson group in each new plc script.

2. crowdin is supposed to be able to recognize when the same English string (display name in this case) has already been translated. @Hamms can you confirm this? this should prevent us from re-translating these strings every time we create a new PL course.

3. my recollection is that the flex categories of a script somehow end up as the module_type on PLC learning modules, which have to be one of these values: https://github.com/code-dot-org/code-dot-org/blob/656517757106252cf864c71894448e947d2cdd64/dashboard/app/models/plc/learning_module.rb#L40 I'm not sure exactly how this will play out, but it's possible you will have to change how this works to get tests passing when you go to move off of flex categories.

**Review by davidsbailey — APPROVED** [review]
Nice work @dmcavoy ! LGTM pending Elijah's approval from the i18n side.

File: dashboard/test/dsl/script_dsl_test.rb
Comment (dmcavoy) [reply dashboard/test/dsl/script_dsl_test.rb]: Thanks Dave for pointing this out again. It is on my radar as something I need to figure out how to deal with. I was considering doing something where if its a PLC scripts then we reserve those module type keys so that you couldn't somehow make a new key-display_name pair for those reserved keys.

Also I confirmed with Elijah that if a translation already exists Crowdin is smart about applying it to new instances of that same string.

Comment (davidsbailey) [reply dashboard/test/dsl/script_dsl_test.rb]: hah, thank you for understanding, I had forgotten pointing this out before :-) I think a special case for the PLC course types sounds great.

Comment (Hamms) [reply dashboard/test/dsl/script_dsl_test.rb]: note you can add a default to the `I18n.t` call in `localized_display_name`, for those situations where the string hasn't yet been persisted to scripts.en.yml

**Review by Hamms — APPROVED** [review]
one nit about defaults, but otherwise looks great!

File: dashboard/test/dsl/script_dsl_test.rb
Comment (dmcavoy) [reply dashboard/test/dsl/script_dsl_test.rb]: Dealt with in this PR: https://github.com/code-dot-org/code-dot-org/pull/34348

**Pull request merged**
